src/main/java/com/limechain/trie/structure/TrieNodeIndex.java

```diff
@@ -0,0 +1,24 @@
+package com.limechain.trie.structure;
+
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.Setter;
+
+//TODO:
+// Consider making this data class an inner class of TrieStructure to guarantee
+// that the index has been generated by the same TrieStructure instance (with a runtime check, ofc)
+
+/**
+ * A semantic wrapper for an integer, representing a node index within our trie structure.
+ * Cannot be instantiated outside, so only the {@link TrieStructure} can generate them and provides them to the user
+ * as an opaque index for nodes.
+ */
+@Data
+public final class TrieNodeIndex {
+    @Setter(AccessLevel.NONE)
+    private int value;
+
+    TrieNodeIndex(int value) {
+        this.value = value;
+    }
+}
```

Comment:
If you make value `final` you can remove the constructor since `@Data` contains `@RequiredArgsConstructor`

Reply:
Good catch, that was part of my uncommited small fixes from Monday. :D